jvm/ml4ir-inference/src/main/scala/ml4ir/inference/tensorflow/data/StringMapHelpers.scala

```diff
@@ -0,0 +1,48 @@
+package ml4ir.inference.tensorflow.data
+
+import scala.collection.JavaConverters._
+
+import java.util.{List => JList, Map => JMap}
+import java.util.function.{Function => JFunction}
+
+import com.google.common.collect.ImmutableMap
+
+import scala.io.Source
+
+case class StringMapQueryAndPredictions(queryContext: JMap[String, String],
+                                        docs: JList[JMap[String, String]],
+                                        predictedScores: Array[Float])
+
+object StringMapCSVLoader {
+
+  def loadDataFromCSV(dataPath: String, featureConfig: ModelFeaturesConfig): Iterable[StringMapQueryAndPredictions] = {
+    val lines = Source.fromFile(dataPath).getLines().toList
+    val (header, dataLines) = (lines.head, lines.tail)
+    val colNames = header.split(",").map(_.replaceAll("\"", ""))
+    val lineMapper: String => Map[String, String] = (line: String) =>
+      colNames.zip(line.split(",")
```

Comment:
I know where you are here! 🙂
We had this formatting issue on the previous integration test. It looks to me that using this string helper is finally the more flexible solution, and I'm all in to have it merge. 

Do you agree? Or would you prefer fixing the Python formatting again?

Reply:
having CSV files with `b'foo bar'` in them are... ugly.  It feels like that should change.  Is that easy, @lastmansleeping ?

Reply:
I did this in the `classification_model.py` ... We end up with [this](https://github.com/salesforce/ml4ir/pull/102/files#r561292870) ... Not a huge fan

Reply:
and re: this being an integration test, @ducouloa - this code assumes the data is already checked in to git, and it probably shouldn't be - it needs your code where we first run the python side (from the exact same git state as the jvm side that's being tested!) to train the model and produce outputs, then compare.  We don't want it to check against a static set.

Also: we want an easily parametrized test harness: someone says "I give python code all of: (test.csv, feature_config.yaml, model_config.yaml, other command-line args), it gives me (model_bundle, model_predictions.csv), then we hand that to JVM, which verifies that the JVM can take all of those and compare model_predictions.csv from the two sides for that config + model + data".  This should be runnable for any set of configs, both at PR-merge time, as well as on production models on real data (i.e. done by / after the production model training completes).

Reply:
yeah, I hear ya - that change looks ugly, I agree.  It's because we're not using cross-platform standard serialization.  I know everyone is complaining about protobuf, but something like a standard avro, protobuf, yaml, json format would be nice, rather than pandas_df.to_csv()...